lib/api/controllers/baseController.js

```diff
@@ -92,6 +99,248 @@ class BaseController {
+    }
+
+    const numberParam = parseInt(rawNumberParam, 10);
+
+    if (isNaN(numberParam)) {
+      errorsManager.throw('invalid_param_type', paramPath, rawNumberParam, 'Number');
+    }
+
+    return numberParam;
+  }
+
+
+  /**
+   * Returns the index specified in the request
+   *
+   * @param {Request} request
+   *
+   * @throws
+   * @returns {String}
+   */
+  getIndex (request) {
+    const index = request.input.resource.index;
+
+    if (! index) {
+      errorsManager.throw('missing_index');
+    }
+
+    return index;
+  }
+
+  /**
+   * Returns the index and collection specified in the request
+   *
+   * @param {Request} request
+   *
+   * @throws
+   * @returns {Object} { index, collection }
+   */
+  getIndexAndCollection (request) {
+    const
+      index = this.getIndex(request),
+      collection = request.input.resource.collection;
+
+    if (! collection) {
+      errorsManager.throw('missing_collection');
+    }
+
+    return { index, collection };
+  }
+
+  /**
+   * Returns the id specified in the request
+   *
+   * @param {Request} request
+   *
+   * @throws
+   * @returns {String}
+   */
+  getId (request) {
+    const id = request.input.resource._id;
+
+    if (! id) {
+      errorsManager.throw('missing_id');
+    }
+
+    if (typeof id !== 'string') {
+      errorsManager.throw('wrong_id_type');
+    }
+
+    return id;
+  }
+
+  /**
+   * Returns the current user id
+   *
+   * @param {Request} request
+   *
+   * @returns {String}
+   */
+  getUserId (request) {
+    return _.get(request, 'context.user._id');
+  }
+
+  getSearchParams (request) {
+    const
+      from = this.getNumberParam(request, 'args.from', 0),
+      size = this.getNumberParam(request, 'args.size', 10),
+      scrollTTL = this.getScrollTTLParam(request),
+      query = this.getObjectParam(request, 'body.query', {}),
+      searchBody = this.getObjectParam(request, 'body', {});
+
+    return { from, size, scrollTTL, query, searchBody };
+  }
+
+  /**
+   * Extract string scroll ttl param from the request or returns undefined
+   *
+   * @param {Request} request
+   *
+   * @returns {String|undefined}
+   */
+  getScrollTTLParam (request) {
+    const scrollTTLParam = request.input.args.scroll;
+
+    if (scrollTTLParam && typeof scrollTTLParam !== 'string') {
+      errorsManager.throw(
+        'invalid_param_type',
+        'args.scroll',
+        scrollTTLParam,
+        'String');
+    }
+
+    return scrollTTLParam;
+  }
+
+  /**
+   * Throws if the body contain one of the specified attribute
+   *
+   * @param {Request} request
+   * @param  {...any} paths
+   */
+  assertBodyHasNotAttributes (request, ...paths) {
+    for (const path of paths) {
+      if (_.get(request.input.body, path)) {
+        errorsManager.throw('must_not_specify_body_attribute', path);
+      }
+    }
+  }
+
+  /**
+   * Throws if the strategy does not exists
+   *
+   * @todo move this method in some kind of "Security" class
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)